plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -196,53 +222,75 @@ internal class LocationPuckManager(
         }
       }
       is LocationPuck3D -> {
-        val modelScaleConstant = 2.0.pow(MAX_ZOOM - MIN_ZOOM)
-        val modelScaleExpression = puck.modelScaleExpression
-        val scaleExpression = if (modelScaleExpression == null) {
+        locationLayerRenderer.styleScaling(get3DPuckScaleExpression(puck, lastMercatorScale))
+      }
+    }
+  }
+
+  private fun get3DPuckScaleExpression(puck: LocationPuck3D, mercatorScale: Double = 1.0): Value {
```

Comment:
from what I see `mercatorScale` is always specified, no need for default parameter